### ISSUE TITLE
addWidgetOverlay not working as expected

### ISSUE THREAD

**Issue by BoHellgren** [flame-engine/flame#265]
I extended the addWidgetOverlay test program a bit, see below. Each time you press the FAB, a new instance of the ExampleGame class is created and its widget is displayed.

The first time you press the FAB, it works as expected. The "Paused" box is displayed on tap.

But after the second time you press the FAB, the "Paused" box is never shown. The console messages are the same, but that's all.

Seems addWidgetOverlay only works for the first Game class instance. This is very annoying and makes this function unusable in my app.

I am using Flame 0.18.3 and Flutter 1.12.13+hotfix.8.

Bo Hellgren
```
import 'package:flutter/material.dart';
import 'package:flame/game.dart';
import 'package:flame/gestures.dart';
import 'package:flame/palette.dart';

class ExampleGame extends Game with HasWidgetsOverlay, TapDetector {
  bool isPaused = false;

  @override
  void update(double dt) {}

  @override
  void render(Canvas canvas) {
    canvas.drawRect(const Rect.fromLTWH(100, 100, 100, 100),
        Paint()..color = BasicPalette.white.color);
  }

  @override
  void onTap() {
    if (isPaused) {
      removeWidgetOverlay('PauseMenu');
      isPaused = false;
    } else {
      print('adding overlay');
      addWidgetOverlay(
          'PauseMenu',
          Center(
            child: Container(
              width: 100,
              height: 100,
              color: const Color(0xFFFF0000),
              child: const Center(child: const Text('Paused')),
            ),
          ));
      isPaused = true;
    }
  }
}

void main() {
  // runApp(ExampleGame().widget);
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key}) : super(key: key);
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  ExampleGame _myGame;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          title: Text('Testing addingOverlay'),
        ),
        body: _myGame == null ? Text('Wait') : _myGame.widget,
        floatingActionButton: FloatingActionButton(
          onPressed: () => newGame(),
          child: Icon(Icons.add),
        ));
  }

  void newGame() {
    print('New game created');
    _myGame = new ExampleGame();
    setState(() {});
  }
}

**Comment by erickzanardo** [comment]
This is very weird, I can't think of any reason why your example is not working as expected, but I will take a look on it.

**Comment by erickzanardo** [comment]
@BoHellgren have you tested on both debug and release mode?

**Comment by BoHellgren** [comment]
Yes, makes no difference.

**Comment by erickzanardo** [comment]
Should be fixed on #267

**Comment by erickzanardo** [comment]
live on `0.19.0`
